Implement third party notification support within alerts

relvira opened the pull request:
Added necessary structs to support third party notifications (via slack, pagerduty, etc) in alerts. This feature is lacking in your new-ish terraform provider and would be of great use. I used your API docs as reference: https://developer.thousandeyes.com/v6/alerts/#/alert-rule-create

I suppose a follow-up pull request on your provider using a newer version of this SDK will need to happen later. Let me know if anything else is needed and if I can facilitate any work to speed things up.

Thanks for your work. 🙇🏼

sfreitas-te commented:
Thanks @relvira for your PR we will review as soon as possible

pedro-te reviewed (commented):
Thank you very much for your contribution @relvira, this is much appreciated! 🙏🏻 
There are some small changes needed after testing your code, which I have written down as part of the review. 

Additionally, could you also add some tests to ensure the marshalling and unmarshalling of the new `NotificationThirdParty` struct works as expected?

Other than that, it looks good 👍🏻

relvira commented:
Hey @pedro-te! 👋🏼 thanks for the review. I have updated the code with recommendations and added a test case for this, hope it all makes sense.

Let me know if there is anything else needed.

pedro-te commented on alert_test.go:
```diff
@@ -95,6 +95,59 @@ func TestClient_CreateAlertRule(t *testing.T) {
+
+	var client = &Client{APIEndpoint: server.URL, AuthToken: "foo"}
+	u := AlertRule{
+		RuleID:   Int64(1),
```
```suggestion
```
Small nitpick here, the ThousandEyes API returns the value for `RuleID`, so in reality this field should not be populated here, but it should be populated in the `expected` variable, which you did. So this just needs to be removed.

relvira replied:
fixed the nit and squashed the commits @pedro-te. Thank you for reviewing this promptly. Once this is done I'll make a follow up contribution to your provider repo bumping the version of the sdk. 

Cheers! 🍻

pedro-te reviewed (commented):
The rest looks good and it works well. Just another small ask: Can you squash the commits into a single one, whose message follows the _conventional commit_ format? So the commit message should say:

> feat: Implement third party notification support within alerts

Once this is updated we can merge the PR. Thanks again @relvira , your contribution is much appreciated

pedro-te commented on alerts.go:
```diff
@@ -36,9 +36,22 @@ type NotificationEmail struct {
 
+// NotificationThirdParty - Alert Rule Notification ThirdParty structure
+type NotificationThirdParty struct {
+	IntegrationId   *string `json:"integrationId,omitempty"`
```
```suggestion
	IntegrationID   *string `json:"integrationId,omitempty"`
```

Ahh nice catch by the linter, this should be `IntegrationID` :)

pedro-te reviewed (changes requested):
Apologies @relvira , this small detail slipped by.

relvira commented:
Oops! good 👁️ - should be fixed and squashed now @pedro-te.

pedro-te commented:
> Oops! good 👁️ - should be fixed and squashed now @pedro-te.

Just missing the tests now 😅 They're still using `IntegrationId`, we should be good to go after that 👍🏻

relvira commented:
ah you're right. I think all should be good now, I've run tests locally and they are 🆗. Thanks again @pedro-te!

pedro-te reviewed (approved):
Thank you @relvira ! Looks good now!

relvira commented:
Thank you @sfreitas-te and @pedro-te! will you generate a new release of the SDK with this included? I want to submit a PR to https://github.com/thousandeyes/terraform-provider-thousandeyes bumping the SDK version so we can use this feature in Terraform.

pedro-te commented:
Yes, @relvira , we will!

pedro-te commented:
@relvira v2.2.0 has been released! Apologies for the delay. Looking forward to the terraform provider PR :)

relvira commented:
Wooho! appreciate the quick turnaround to create the release here, @pedro-te. I've just submitted a PR to the provider repo https://github.com/thousandeyes/terraform-provider-thousandeyes/pull/106. Feel free to 👀 when you have some time.

🙇🏼
